hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/HoodieHiveCatalog.java

```diff
@@ -438,8 +439,10 @@ public CatalogBaseTable getTable(ObjectPath tablePath) throws TableNotExistExcep
       LOG.warn("{} does not have any hoodie schema, and use hive table schema to infer the table schema", tablePath);
       schema = HiveSchemaUtils.convertTableSchema(hiveTable);
     }
+    org.apache.flink.table.api.Schema resultSchema = DataTypeUtils.dropIfExistsColumns(schema, HoodieRecord.HOODIE_META_COLUMNS_WITH_OPERATION);
+
```

Comment:
In line419, we already ignore the metadata column, so why drop it again?

Reply:
Indeed, I have removed.